File: server/src/main/java/com/krillsson/sysapi/core/metrics/ProcessesMetrics.java
package com.krillsson.sysapi.core.metrics;

import com.krillsson.sysapi.core.domain.processes.Process;
import com.krillsson.sysapi.core.domain.processes.ProcessesInfo;
import oshi.software.os.OperatingSystem;

import java.util.Optional;

public interface ProcessesMetrics {
    ProcessesInfo processesInfo(OperatingSystem.ProcessSort sortBy, int limit);
    Optional<Process> getProcessByPid(int pid);
}

File: server/src/main/java/com/krillsson/sysapi/core/metrics/ProcessesMetrics.kt
package com.krillsson.sysapi.core.metrics

import com.krillsson.sysapi.core.domain.processes.Process
import com.krillsson.sysapi.core.domain.processes.ProcessSort
import com.krillsson.sysapi.core.domain.processes.ProcessesInfo
import java.util.Optional

interface ProcessesMetrics {
    fun processesInfo(sortBy: ProcessSort, limit: Int): ProcessesInfo
    fun getProcessByPid(pid: Int): Optional<Process>
}